x/twap/logic_test.go

```diff
@@ -1026,25 +1027,11 @@ func (s *TestSuite) TestUpdateRecords() {
 				},
 			},
 
-			expectedHistoricalRecords: []expectedResults{
-				// The original record at t.
-				{
-					spotPriceA: baseRecord.P0LastSpotPrice,
-					spotPriceB: baseRecord.P1LastSpotPrice,
-				},
-				// The new record added.
-				// TODO: it should not be possible to add a record between existing records.
-				// https://github.com/osmosis-labs/osmosis/issues/2686
-				{
-					spotPriceA:   sdk.OneDec(),
-					spotPriceB:   sdk.OneDec().Add(sdk.OneDec()),
-					isMostRecent: true,
-				},
-				// The original record at t + 1.
-				{
-					spotPriceA: tPlus10sp5Record.P0LastSpotPrice,
-					spotPriceB: tPlus10sp5Record.P1LastSpotPrice,
-				},
+			expectError: types.InvalidRecordTimeError{
```

Comment:
Can we also add a test case for having block height greater than the last record's time and checking the error?

Reply:
Yes of course